agent/api/task/service_connect_test.go

```diff
@@ -0,0 +1,261 @@
+//go:build unit
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//    http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package task
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/amazon-ecs-agent/agent/acs/model/ecsacs"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	awsvpcMode                      = "awsvpc"
+	bridgeMode                      = "bridge"
+	testAwsvpcPortOverride          = "8080"
+	testAwsvpcPortDefault           = "8080"
+	testBridgePortOverride          = "8080"
+	testBridgePortDefault           = "15000"
+	testBridgeHostPort              = "8080"
+	testServiceConnectContainerName = "ecs-service-connect"
+	testStatsConfigPath             = "testPath"
+	testHostName                    = "testHostName"
+	testAddress                     = "testAddress"
+	testOutboundListenerName        = "testOutboundListener"
+	testInboundListenerName         = "testInboundListener"
+	testIPV4Cidr                    = "testIPV4Cidr"
+	testIPV6Cidr                    = "testIPV6Cidr"
+)
+
+func getTestACSTaskAttachment(networkMode string, override bool) *ecsacs.Task {
+	ingressConfig := ""
+	switch networkMode {
+	case awsvpcMode:
+		if override {
+			ingressConfig = fmt.Sprintf(`\"listenerPort\":%s`, testAwsvpcPortOverride)
+		} else {
+			ingressConfig = fmt.Sprintf(`\"interceptPort\":%s,\"listenerName\":\"%s\"`, testAwsvpcPortDefault, testInboundListenerName)
+		}
+	case bridgeMode:
+		if override {
+			ingressConfig = fmt.Sprintf(`\"listenerPort\":%s,\"hostPort\":%s`, testBridgePortOverride, testBridgeHostPort)
+		} else {
+			ingressConfig = fmt.Sprintf(`\"listenerPort\":%s`, testBridgePortDefault)
+		}
+	}
+
+	testAWSVPCServiceConnectConfig := fmt.Sprintf(`"{\"statsConfig\":{\"path\":\"%s\"},\"egressConfig\":{\"listenerName\":\"%s\",\"vip\":{\"ipv4Cidr\":\"%s\",\"ipv6Cidr\":\"%s\"}},\"dnsConfig\":[{\"hostname\":\"%s\",\"address\":\"%s\"}],\"ingressConfig\":[{%s}]}"`, testStatsConfigPath, testOutboundListenerName, testIPV4Cidr, testIPV6Cidr, testHostName, testAddress, ingressConfig)
```

Comment:
nit: could we use a multi-line string here? kinda painful to read like this :)

Reply:
test service config is broken down into multi-lines, and couple format constants, shown as follows, are defined and used in commit 8e69368bc5c1d7bd3ab71391103f4b6a69f06054.  
```
testStatsConfigFormat                             = `\"statsConfig\":{\"path\":\"%s\"}`
testEgressConfigFormat                            = `\"egressConfig\":{\"listenerName\":\"%s\",\"vip\":{\"ipv4Cidr\":\"%s\",\"ipv6Cidr\":\"%s\"}}`
testDnsConfigFormat                               = `\"dnsConfig\":[{\"hostname\":\"%s\",\"address\":\"%s\"}]`
testIngressConfigAwsVpcDefaultFormat              = `\"ingressConfig\":[{\"interceptPort\":%s,\"listenerName\":\"%s\"}]`
testIngressConfigBridgeOverrideFormat             = `\"ingressConfig\":[{\"listenerPort\":%s,\"hostPort\":%s}]`
testIngressConfigListenerPortOnlyFormat           = `\"ingressConfig\":[{\"listenerPort\":%s}]`
testInvalidIngressConfigAwsVpcDefaultFormat       = `\"ingressConfig\":[{\"interceptPort\":%s,\"listenerName\":\"%s\"}` + `,{\"interceptPort\":%s,\"listenerName\":\"%s\"}]`
testInvalidIngressConfigListenerPortFormat        = `\"ingressConfig\":[{\"listenerPort\":%s},{\"listenerPort\":%s}]`
testInvalidIngressConfigBridgeOverrideFormat      = `\"ingressConfig\":[{\"listenerPort\":%s,\"hostPort\":%s},{\"listenerPort\":%s,\"hostPort\":%s}]`
testInvalidMixedIngressConfigBridgeOverrideFormat = `\"ingressConfig\":[{\"listenerPort\":%s,\"hostPort\":%s},{\"interceptPort\":%s,\"listenerName\":\"%s\"}]`
```
Thanks.